airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_stream.py

```diff
@@ -143,8 +146,7 @@ def stream_slices(
 
         :param sync_mode:
         :param cursor_field:
-        :param stream_state:
+        :param stream_state: we knowingly avoid using stream_state as we want cursors to manage their own state
         :return:
         """
-        # this is not passing the cursor field because it is known at init time
-        return self.retriever.stream_slices(sync_mode=sync_mode, stream_state=stream_state)
+        return self.retriever.stream_slices()
```

Comment:
If `retriever.stream_slices` only accepts an instance of a class, do we require any additional arguments to be awaited in the `stream_slices` method of the `DeclarativeStream` class?

Reply:
The `Stream` interface [requires those parameters to be there](https://github.com/airbytehq/airbyte/blob/3df819612ffa9ab34803483de69b9578cac92cf7/airbyte-cdk/python/airbyte_cdk/sources/streams/core.py#L206). This is [the contract between the AbstractSource and the Stream](https://github.com/airbytehq/airbyte/blob/3df819612ffa9ab34803483de69b9578cac92cf7/airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py#L245-L248). This is why we need to keep the parameters on the `DeclarativeStream` level but we won't keep it. Does that make sense?